### PR TITLE
Issue622 wet coil refactor 3

### PR DESCRIPTION
**Pull request by AntoineGautier**
I create this draft PR to enable inline comments on the code.

### CODE REVIEW

**Comment by AntoineGautier** [comment]
TODO 

- [ ] Propagate `allowFlowReversal{1, 2}` to `heaCoo` and `heaCooHum_u`
- [ ] Fix the issue in `Buildings.Fluid.HeatExchangers.Examples.WetCoilEffectivenessNTU` that exhibits convergence issues if  `allowFlowReversal1=true`.

**Comment by AntoineGautier** [comment]
@kim1077 I just committed the version with the IBPSA wet coil model (from https://github.com/ibpsa/modelica-ibpsa/pull/1113). The example and validation models have also been refactored. I will focus on benchmarking the simulation performance on large models. I let you go ahead with benchmarking the model precision.

**Comment by AntoineGautier** [comment]
@kim1077 I included closed loop test cases based on Spawn room models or load time series under `Experimental.DHC.Loads.Examples` in https://github.com/lbl-srg/modelica-buildings/pull/2001/commits/ffa1fe7f8b25818f8715010326a2f18bf71f6f22. There are still some convergence issues with `WetcalcsFuzzy_V2_2_4` that we can discuss tomorrow.

**Comment by AntoineGautier** [comment]
@kim1077 I propose to close this PR without merging and open a new one with the branch `issue622_wetCoil` that I just created with the following steps:

- Merge the master 3c2d2970d44101241a84348eab3963f58a04ec83 into `issue622_wet_coil_refactor_3` 456406ec2c
- From the master 3c2d2970d44101241a84348eab3963f58a04ec83 `git checkout -b issue622_wetCoil`
- `git diff issue622_wet_coil_refactor_3 --name-only > ~/tmp.txt`
- Manually edit `~/tmp.txt` to remove all unnecessary modifications. The resulting file is attached for reference.
- `git checkout issue622_wet_coil_refactor_3 -- $(cat ~/tmp.txt)`
- Rename classes, especially `WetEffectivenessNTU_Fuzzy_V3` is now `WetCoilEffectivenessNTU`.

I removed all other versions of the model and the whole `Buildings.Fluid.HeatExchangers.H_Example` package.
Can you please double check the version that is now on `issue622_wetCoil` because I have some doubt that it is indeed the last version you worked on. Indeed, when I first tried to run `Buildings.Fluid.HeatExchangers.Examples.WetCoilEffectivenessNTU` I got the following error (which is fixed now with my last commit cdb7778). Also the UA_nominal computation is not included but that might be OK for now.

```
Class or component 'TWatOut_init' not found in WetEffectivenessNTU_Fuzzy_V3 hex 
```

After your confirmation I will move forward with reviewing the code on `issue622_wetCoil`.


[tmp.txt](https://github.com/lbl-srg/modelica-buildings/files/5955083/tmp.txt)

**Comment by AntoineGautier** [comment]
@mwetter Can we delete the following branches?
`issue622_wet_coil_refactor_3`
`issue622_wet_coil_refactor_2`
`issue622_wet_coil`

The development will be merged through the new branch `issue622_wetCoil`.

File: Buildings/Fluid/HeatExchangers/BaseClasses/DryCalcsFuzzy_V2_2.mo [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+    x =   dryfra-delta,
+    deltax = delta/2);
+
+  deltaCMin = delta*min(mAir_flow_nominal*cpAir,mWat_flow_nominal*cpWat);
+  deltaCMax = delta*max(mAir_flow_nominal*cpAir,mWat_flow_nominal*cpWat);
+
+  CWat_flow = mWat_flow*cpWat;
+  CAir_flow = mAir_flow*cpAir;
+  //CMin_flow = min(CWat_flow, CAir_flow);
+   CMin_flow =Buildings.Utilities.Math.Functions.smoothMin(
```
[the remark this answers is not in the document]
Comment (kim1077) [reply Buildings/Fluid/HeatExchangers/BaseClasses/DryCalcsFuzzy_V2_2.mo]: @AntoineGautier, Did you review V2_2? The latest codes end with "V3" which utilized epsilon_C.

File: Buildings/Fluid/HeatExchangers/BaseClasses/DryCalcsFuzzy_V2_2.mo [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+  Z = CMin_flow/CMaxNonZer_flow   "Braun 1988 eq 4.1.10";
+   gai = Buildings.Utilities.Math.Functions.spliceFunction(
+            pos=1,
+            neg=0,
+            x=CMin_flow-deltaCMin,
+            deltax=deltaCMin/2);
+   if (gai <= 0) and (gai >=0) then
+     NTU = 0;
+     eps = 1; // around zero flow, eps=Q/(CMin*dT) should be one
+   else
```
[the remark this answers is not in the document]
Comment (kim1077) [reply Buildings/Fluid/HeatExchangers/BaseClasses/DryCalcsFuzzy_V2_2.mo]: @AntoineGautier Again, this code "V2_2" is an obsolete one (to test and compare multiple approaches). Would you please double-check whether you reviewed "V3"?

File: Buildings/Fluid/HeatExchangers/BaseClasses/DryCalcsFuzzy_V2_2.mo (hunk 1 above)
[the remark this answers is not in the document]
Comment (AntoineGautier) [reply Buildings/Fluid/HeatExchangers/BaseClasses/DryCalcsFuzzy_V2_2.mo]: @kim1077 Please do not take into account the review comments on this PR. They were made a long time ago but somehow Github just asked me to close the review, which made then visible for you on this page just now I believe.

[the remark this answers is not in the document]
Comment (AntoineGautier) [reply Buildings/Fluid/HeatExchangers/BaseClasses/DryCalcsFuzzy_V2_2.mo]: `WetEffectivenessNTU_Fuzzy_V3` is the only version I included in the new branch I intend to review, see https://github.com/lbl-srg/modelica-buildings/pull/2001#issuecomment-776349084.

**Comment by AntoineGautier** [comment]
This is closed without merging. 
The development will be merged through the new branch `issue622_wetCoil`.

**Pull request closed without merging**